src/client/src/consumer.rs

```diff
@@ -565,33 +572,67 @@ static MAX_FETCH_BYTES: Lazy<i32> = Lazy::new(|| {
 });
 
 /// Configures the behavior of consumer fetching and streaming
-#[derive(Debug)]
+#[derive(derive_builder::Builder, Debug)]
 pub struct ConsumerConfig {
+    /// Maximum number of bytes to be fetched at a time.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use fluvio::ConsumerConfig;
+    /// let config = ConsumerConfig::builder()
+    ///     .max_bytes(1000)
+    ///     .build()
+    ///     .unwrap();
+    /// ```
+    #[builder(default = "*MAX_FETCH_BYTES")]
     pub(crate) max_bytes: i32,
+    #[builder(setter(skip))]
     pub(crate) isolation: Isolation,
-    wasm_module: Vec<u8>,
+    /// A WASM module to use as a SmartStream filter
+    #[builder(setter(into, strip_option), default)]
+    smartstream_filter: Option<Vec<u8>>,
```

Comment:
Option makes more sense here IMO. Seeing this in the struct as `smartstream_filter: Vec<u8>` makes me think that the field will always be populated, whereas an `Option<Vec<u8>>` semantically lets me know that I should expect to check whether a value is present or not. Note that this also does not have any runtime cost, because any type containing a pointer (like vec does) inside an Option will be niche-optimized: https://stackoverflow.com/questions/16504643/what-is-the-overhead-of-rusts-option-type